api/app/main.py

```diff
@@ -1,7 +1,46 @@
-from fastapi import FastAPI
+from typing import Optional
+from fastapi import FastAPI, Request, Header
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+import requests
+from PIL import Image
+from io import BytesIO
+import numpy as np
+import matplotlib.pyplot as plt
+import json
 
+#uvicorn main:app --reload
 app = FastAPI()
+templates = Jinja2Templates(directory="./HTML")
 
-@app.get("/")
-def root():
-    return {"hello": "world"}
+@app.get("/", response_class=HTMLResponse)
+def root(request: Request):
+    return templates.TemplateResponse("homepage.html", {"request": request}) #just a useless homepage.XD
+
+@app.get("/image/")
+def url_post(request: Request):
+    return templates.TemplateResponse("showimag.html", {"request": request,"test":request.headers}) #the HTML file can send requests with image url in the headers. using XMLHttpRequests
+
+@app.get("/image/rgbdata/")
+async def read_items(image_url: Optional[str] = Header(None)): #fastapi will transfer the dash from - to _
+    response = requests.get(image_url)
+    image = Image.open(BytesIO(response.content)) #how is the image encoded?
```

Comment:
really nice 🎉 
the image is encoded as bytes - BytesIO just acts as a transport layer between the request object and PIL